src/main/java/io/boomerang/controller/InternalController.java

```diff
@@ -70,14 +70,23 @@ public List<WorkflowShortSummary> getAllWorkflows() {
 
   @GetMapping(value = "/system-workflows")
   @Deprecated
-  public List<WorkflowShortSummary> getAllSystemworkflows() {
+  public List<WorkflowShortSummary> getAllSystemWorkflows() {
     return workflowService.getSystemWorkflowShortSummaryList();
   }
 
   @PutMapping(value = "/workflow/event", consumes = "application/cloudevents+json; charset=utf-8")
-  public ResponseEntity<CloudEventImpl<EventResponse>> acceptEvent(
-      @RequestHeader Map<String, Object> headers, @RequestBody JsonNode payload) {
-    return ResponseEntity.ok(eventProcessor.processHTTPEvent(headers, payload));
+  public ResponseEntity<EventResponse> acceptEvent(@RequestHeader HttpHeaders headers,
+      @RequestBody String payload) {
+    CloudEvent cloudEvent =
+        CloudEventHttpUtils.toReader(headers, () -> payload.getBytes()).toEvent();
+    try {
+      eventingService.processCloudEventRequest(cloudEvent);
```

Comment:
if we make this non blocking as well i.e. CompletableFuture or similar, then I was reading up that we would just respond with a 202. In that the event is accepted but we are still potentially processing.

That appears to be a more common async loosely coupled event approach, than needing to respond to the event.

Reply:
@tlawrie, I would leave this method as blocking since `processCloudEventRequest()` does some validation and verification on the CloudEvent. If something's not looking right, an exception is raised by the service and is further returned as a response to the client invoking this API, providing extra information on what's wrong.

#### Couple of `Exception` examples

- `BoomerangException`
- `InvalidPropertiesFormatException`
- `CloudEventsError`
- Invalid Token
- Other...

Reply:
Hey @corcoja for the internal controller that should be fine. Particularly because its for inbound HTTP based events.

The async was mainly a comment for those that we are creating whilst running a workflow, such as the status events.